08-power-quality-monitor/README.md

```diff
@@ -0,0 +1,303 @@
+
+## Events
+
+A power monitoring event is sent every `heartbeat_mins` minutes or when an alert is sent.
+Events are sent to the notefile `power.qo`, and have this structure in the event body:
+
+```json
+{
+    "current": 0.2846,      // Line RMS current (A)
+    "frequency": 59.8125,   // Line AC frequency (Hz)
+    "power": 7.9,           // Line Power (Watts)
+    "voltage": 118.6,       // Line RMS voltage (V)
+}
+```
+
+Regular monitoring events are not immediately synched to Notehub, but are sent once per hour, as given by the `outbound` property in the `hub.set` request. You can change this behavior by setting the preprocessor symbol `SYNC_POWER_MONITORING_NOTES` in [`app.h`](./firmware/notepower/app.h#L20).
+
+The event body also includes these fields:
+
+* `reactivePower`: The measured reactive power (in VAR).
+* `apparentPower`: The measured apparent power (in VA).
+* `powerFactor`: The power factor - active power divided by apparent power.
+
+> Note: that when a property in an event is zero, or false, it is not present in the event routed to notehub. For more details see [How the Notecard works with JSON](https://dev.blues.io/notecard/notecard-walkthrough/json-fundamentals/#how-the-notecard-works-with-json).
+
+
+### Alerts
+
+When the device detects over or under voltage, current or power, or detects a change in these greater than the configured percentage, a power monitoring event is sent as above, with an additional property `alert` that lists the comma-separated reason(s) for the alert. Depending upon the cause of the alert, you may see one or more of these values present:
+
+* `undervoltage`, `overvoltage`: the measured RMS voltage is not within the bounds set by the environment variables `alert_under_voltage` and `alert_over_voltage`.
+
+* `voltage`: the measured voltage changed by more than the percent specified in the environment variable `alert_change_voltage_percent`.
+
+* `undercurrent`, `overcurrent`: the measured RMS current is not within the bounds set by the environment variables `alert_under_current_amps`, `alert_over_current_amps`.
+
+* `current`: the measured current changed by more than the percent specified in the environment variable `alert_change_current_percent`.
+
+* `underpower`, `overpower`: the measured active power is not within the bounds set by the environment variables  `alert_under_power_watts` and `alert_over_power_watts`.
+
+* `power`: the measured active power changed by more than the percent specified in the environment variable `alert_change_power_percent`.
+
+When an alert is triggered, it is immediately synched to Notehub.
+
+
+## Routing Data out of Notehub
+
+Notehub supports forwarding data a wide range of API endpoints by using the Route feature. This can be used to forward your power monitoring data to external dashboards and alerts to a realtime notification service.  Here we will use Twilio SMS API to send a notification to a phone number.
```

Comment:
```suggestion
Notehub supports forwarding data a wide range of API endpoints by using the Route feature. This can be used to forward your power monitoring data to external dashboards and alerts to a realtime notification service.  Here, we will use Twilio SMS API to send a notification to a phone number.
```